### PR TITLE
Add solution for Richest Customer Wealth

### LINKED ISSUE #14: Richest Customer Wealth
## Issue Overview
Implement a solution for the "Richest Customer Wealth" LeetCode problem to practice working with 2D arrays and aggregation operations in Rust.

## Feature Details
Create a Rust implementation for Richest Customer Wealth.

Problem summary:
You are given an m x n integer grid accounts where accounts[i][j] is the amount of money the i-th customer has in the j-th bank. Return the wealth that the richest customer has, where a customer's wealth is the amount of money they have in all their bank accounts.

Constraints:
- m == accounts.length
- n == accounts[i].length
- 1 <= m, n <= 50
- 1 <= accounts[i][j] <= 100

The solution should:
- Handle 2D arrays/vectors efficiently in Rust
- Utilize appropriate iteration methods and aggregation functions
- Include proper error handling for edge cases
- Implement comprehensive tests

## Expected Outcome
A working Rust implementation that correctly identifies the maximum wealth with clean, efficient code, proper error handling, and thorough test coverage.

### LINKED ISSUE #23: How Many Numbers Are Smaller Than the Current Number
## Issue Overview
Implement a solution for the "How Many Numbers Are Smaller Than the Current Number" LeetCode problem to practice array comparison algorithms in Rust.

## Feature Details
Create a Rust implementation for the problem: https://leetcode.com/problems/how-many-numbers-are-smaller-than-the-current-number/description/

Problem summary:
Given an array of integers `nums`, for each `nums[i]` find out how many numbers in the array are smaller than it. Return the answer in an array.

Constraints:
- 2 <= nums.length <= 500
- 0 <= nums[i] <= 100

The solution should:
- Consider both naive and optimized approaches
- Demonstrate array traversal and comparison techniques
- Include proper error handling
- Feature comprehensive tests for various scenarios

## Expected Outcome
A well-documented Rust implementation that correctly counts smaller numbers for each array element, with proper error handling and thorough test coverage, focusing on both correctness and efficiency.

### PR DESCRIPTION
**Pull request by zmrdltl**
Related Issues
closes #14 

PR Description
Adds a solution for the LeetCode "Richest Customer Wealth" problem.

Key features:
- Functional approach using iterator methods (map, max)
- Proper error handling with ok_or for Option conversion
- Input validation for empty arrays
- Comprehensive test coverage for examples and edge cases

This implementation follows Rust best practices with immutable variables and expressive iterator chains.

### CODE REVIEW

File: src/vectors/richest_customer_wealth.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    if accounts.iter().any(Vec::is_empty) {
+        return Err("Input array cannot contain empty arrays");
+    }
+
+    accounts
+        .iter()
+        .map(|account| account.iter().sum())
+        .max()
+        .ok_or("Failed to calculate maximum wealth")
```
Comment (kimhanbeom) [inline src/vectors/richest_customer_wealth.rs:42]: `max()`가 `None`인 경우는 `iterator`가 empty인 경우인데  `if accounts.is_empty()`에서 이미 empty를 체크하므로  max의 결과는 항상 Some일 것입니다. 그렇기 때문에 해당 부분을`ok_or` 대신 `expect`로 변경해주세요.

**Review by kimhanbeom — CHANGES_REQUESTED** [review]

**Comment by zmrdltl** [comment]
# PR Management Update: Restructuring PRs #41 and #42

## Current Situation

Pull Requests #41 (Richest Customer Wealth) and #42 (Shuffle the Array) have been incorrectly structured due to branch dependency issues. Current structure:

```
main
└── richest-customer-wealth (#41)
    └── shuffle-the-array (#42)
```

This structure has resulted in #41 being automatically closed when #42 was merged first.

## Correction Plan

To address this issue, I will restructure the PRs as follows:

1. Close current PRs #41 and #42.
2. Create independent branches for each feature, directly branched from main:

```
main
├── richest-customer-wealth-new #41 
└── shuffle-the-array-new #42
```

3. Open new independent PRs from these branches.

## Action Items

- I will create new PRs today and request reviews.
- The new PRs will incorporate all code changes and feedback from the existing PRs.
- With this independent branch structure, each PR can be reviewed and merged independently without affecting others.

## Note

This experience has highlighted the importance of using independent branches directly from main for all feature development. This approach prevents PR dependency issues and ensures clearer code review and merge processes.

I apologize for any confusion caused. I will improve branch management practices for more efficient collaboration going forward.
